src/app/modules/twenty-four-hour-time/twenty-four-hour-time.component.html

```diff
@@ -0,0 +1,168 @@
+      (blur)="blurCallBackFunction($event)"
+      (keyup)="keyupCallBackFunction($event)"
+      (focus)="addFocusStyling()"
+    />
+  </div>
+  <div class="ds-l-col--2 clock-parent">
+    <span
+      role="button"
+      tabindex="0"
+      alt="clock"
+      class="clock-image"
+      aria-label="Activate enter to open the time picker."
+      (click)="openTimePicker($event)"
+      (keyup.enter)="openTimePicker($event)"
+    ></span>
+  </div>
+</div>
+<div class="time-picker
+    ds-u-radius
+    ds-u-border--1
+    ds-u-border--dark"
+  *ngIf="timePickerVisible"
+  id="time-picker-{{id}}"
+  (click)="stopPropagation($event)"
+  (focusout)="focusLost($event)"
+>
+  <div class="time-picker-header">
+    <div tabindex="-1" class="time-picker-label">Select time</div>
+    <span
+      id="{{id}}_close_time_picker"
+      tabindex="0"
+      alt="close"
+      class="close-image"
+      aria-label="Activate enter to close the time picker"
+      (click)="closeTimePicker($event)"
+      (keyup.enter)="closeTimePicker($event)"
+    ></span>
+  </div>
+  <div class="time-picker-body" tabindex="-1">
+    <div class="other-number-row">
+      <div class="other-number-column other-column-top">
+        <div
+          class="bumper-icon top-icon"
+          tabindex="0"
+          aria-label="Activate enter to decrement the hour value"
+          role="button"
+          (click)="stepTime('hours', 'decrement')"
+          (keyup.enter)="stepTime('hours', 'decrement')"
+        >
+          <fa-icon [icon]="faChevronUp"></fa-icon>
+        </div>
+        <div *ngFor="let hour of previousHours">
+          {{hour}}
+        </div>
+      </div>
+      <div class="other-number-spacer"></div>
+      <div class="other-number-column other-column-top">
+        <div
+          class="bumper-icon top-icon"
+          tabindex="0"
+          aria-label="Activate enter to decrement the minute value"
+          role="button"
+          (click)="stepTime('minutes', 'decrement')"
+          (keyup.enter)="stepTime('minutes', 'decrement')"
+        >
+          <fa-icon [icon]="faChevronUp"></fa-icon>
+        </div>
+        <div *ngFor="let minute of previousMinutes">
+          {{minute}}
+        </div>
+      </div>
+    </div>
+    
+    <div class="selected-time-row">
+      <div
+        class="item-selected"
+      >{{timeHours}}</div>
+      <div class="time-colon">:</div>
+      <div
+        class="item-selected"
+      >{{timeMinutes}}</div>
+    </div>
+
+    <div class="other-number-row">
+      <div class="other-number-column other-column-bottom">
+        <div *ngFor="let hour of nextHours">
+          {{hour}}
+        </div>
+        <div
+          class="bumper-icon bottom-icon"
+          tabindex="0"
+          aria-label="Activate enter to increment the hour value"
+          role="button"
+          (click)="stepTime('hours', 'increment')"
+          (keyup.enter)="stepTime('hours', 'increment')"
```

Comment:
Ok... but you specifically watch for only `keyup.enter` ... `keyup.space` [should also work](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#keyboard_interactions)

... that said, seems it's recommended to use <button> or `<input type="button">` instead... which would perhaps let `onclick` [do all the lifting](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#required_javascript_features) of `click` and `keyup` and needing to strictly match with the `aria-label` 
```
For native HTML <button> elements, the button's onclick event fires for mouse clicks and when the user presses Space or Enter while the button has focus.
```

Reply:
This is something that I struggled with for like a day or so: every time I hit space it would scroll the entire page as well, even when preventing default or stopping propagation. I noticed that a few of our other components and various sites around the web also only seem to respond to enter: not really a great rule but an in-the-moment judgment call I made 😬 

Are you okay if I add it to our existing list of future improvements?